lib/lines.js

```diff
@@ -448,6 +448,22 @@ Lp.guessTabWidth = function() {
     return secret.cachedTabWidth = result;
 };
 
+// Determine if the list of lines has a first line that's empty of non-comment
+// tokens. If this is the case, the code may need to be wrapped in parens to
+// avoid ASI issues. Calling code should be more defensive if we return true,
+// so false positives are ok.
+Lp.isFirstLineOnlyComment = function () {
+    var secret = getSecret(this);
+    if (secret.infos.length === 0) {
+        return false;
+    }
+    var firstLineInfo = secret.infos[0],
+        firstLine = firstLineInfo.line.trim();
```

Comment:
This method has some (admittedly non-obvious) problems. For example,
* The `secret.infos[i].line` string may have been inherited from another `Lines` object, so it needs to be sliced from `secret.infos[i].sliceStart` to `secret.infos[i].sliceEnd`, then trimmed.
* What if `firstLine.startsWith("/*")` but then later has a `*/` followed by non-comment tokens?

That said, I think we can do without the method, so no need to fix it.